ps_featuredproducts.php

```diff
@@ -263,37 +314,94 @@ public function getWidgetVariables($hookName = null, array $configuration = [])
+				' . Product::sqlStock('p', 0) . '
+				LEFT JOIN `' . _DB_PREFIX_ . 'category_lang` cl
+					ON (product_shop.`id_category_default` = cl.`id_category`
+					AND cl.`id_lang` = ' . (int) $this->context->language->id . Shop::addSqlRestrictionOnLang('cl') . ')
+				LEFT JOIN `' . _DB_PREFIX_ . 'product_lang` pl
+					ON (p.`id_product` = pl.`id_product`
+					AND pl.`id_lang` = ' . (int) $this->context->language->id . Shop::addSqlRestrictionOnLang('pl') . ')
+				LEFT JOIN `' . _DB_PREFIX_ . 'image_shop` image_shop
+					ON (image_shop.`id_product` = p.`id_product` AND image_shop.cover=1 AND image_shop.id_shop=' . (int) $this->context->shop->id . ')
+				LEFT JOIN `' . _DB_PREFIX_ . 'image_lang` il
+					ON (image_shop.`id_image` = il.`id_image`
+					AND il.`id_lang` = ' . (int) $this->context->language->id . ')
+				LEFT JOIN `' . _DB_PREFIX_ . 'manufacturer` m
+					ON m.`id_manufacturer` = p.`id_manufacturer`
+                LEFT JOIN (
+                    SELECT od.product_id, count(od.id_order) as orders_placed
+                    FROM `' . _DB_PREFIX_ . 'order_detail` od
+                    LEFT JOIN `' . _DB_PREFIX_ . 'orders` o
+                        ON o.`id_order` = od.`id_order`
+                    GROUP BY od.product_id
+                ) o ON o.product_id = p.id_product
+				WHERE product_shop.`id_shop` = ' . (int) $this->context->shop->id . '
+					AND cp.`id_category` = ' . (int) $idCategory . '
+                    AND product_shop.`active` = 1
+                    AND product_shop.`visibility` IN ("both", "catalog")'
+                    . ($onlyAvailable ? ' 
+                    AND (
+                        (stock.out_of_stock = 2 AND ' . ($outOfStockGlobal == 0 ? 'IFNULL(stock.quantity, 0) > 0' : 'TRUE') . ')
+                        OR (stock.out_of_stock = 1)
+                        OR (stock.out_of_stock = 0 AND IFNULL(stock.quantity, 0) > 0)
+                    )'
+                    : '');
+
+        if ($random) {
+            if ($popularFirst) {
+                $sql .= ' ORDER BY -LOG(1.0 - RAND()) / (IFNULL(o.orders_placed, 0.1)) ';
+            } else {
+                $sql .= ' ORDER BY RAND() ';
+            }
         } else {
-            $query->setSortOrder(new SortOrder('product', 'position', 'asc'));
+            if ($popularFirst) {
+                $sql .= ' ORDER BY o.orders_placed DESC, cp.`position` ASC ';
+            }
+            else {
+                $sql .= ' ORDER BY cp.`position` ASC ';
+            }
+        }
+
+        $sql .= ' LIMIT ' . (int) $nProducts;
+
+        $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($sql, true, false);
+
+        if (!$result) {
+            $result = [];
         }
 
-        $result = $searchProvider->runQuery(
-            $context,
-            $query
-        );
+        // Modify SQL result
+        $products = Product::getProductsProperties($this->context->language->id, $result);
```

Comment:
@kelostrada again, it is inside Presenter, the only thing you need for Presenter to work are product IDs